libcudacxx/include/cuda/std/__concepts/__concept_macros.h

```diff
@@ -323,6 +339,74 @@
 #    define _LIBCUDACXX_TRAILING_REQUIRES(...)      ->_Concept::_Requires_t < __VA_ARGS__ _LIBCUDACXX_TRAILING_REQUIRES_AUX_
 #  endif
 
+////////////////////////////////////////////////////////////////////////////////
+// _LIBCUDACXX_REQUIRES_EXPR
+// Usage:
+//   template <typename T>
+//   _LIBCUDACXX_CONCEPT equality_comparable =
+//     _LIBCUDACXX_REQUIRES_EXPR((T), T const& lhs, T const& rhs) (
```

Comment:
This has always the same structure

Template arguments -> `_LIBCUDACXX_CONCEPT` -> name -> `_LIBCUDACXX_REQUIRES_EXPR` -> template arguments -> ...

This makes me wonder whether it should be a single macro rather than two

Reply:
`_LIBCUDACXX_CONCEPT` is useful independently of `_LIBCUDACXX_REQUIRES_EXPR`. they should be kept separate. i find 20 more uses of `_LIBCUDACXX_CONCEPT` than `_LIBCUDACXX_CONCEPT_FRAGMENT`.